pc/io_test.go

```diff
@@ -72,3 +74,78 @@ func TestUnmarshalAscii(t *testing.T) {
 		lt.Incr()
 	}
 }
+
+func TestMarshal(t *testing.T) {
+	t.Run("CheckViewPoint", func(t *testing.T) {
+		testCases := map[string]struct {
+			pp             *PointCloud
+			expectedHeader PointCloudHeader
+		}{
+			"NoViewpointProvided": {
+				pp: &PointCloud{
+					PointCloudHeader: PointCloudHeader{
+						Fields: []string{"x", "y", "z"},
+						Size:   []int{4, 4, 4},
+						Count:  []int{1, 1, 1},
+						Type:   []string{"F", "F", "F"},
+						Width:  3,
+						Height: 1,
+					},
+					Points: 3,
+					Data:   make([]byte, 3*4*3),
+				},
+				expectedHeader: PointCloudHeader{
+					Fields:    []string{"x", "y", "z"},
+					Size:      []int{4, 4, 4},
+					Count:     []int{1, 1, 1},
+					Type:      []string{"F", "F", "F"},
+					Width:     3,
+					Height:    1,
+					Viewpoint: []float32{0, 0, 0, 1, 0, 0, 0},
+				},
+			},
+			"ViewpointProvided": {
+				pp: &PointCloud{
+					PointCloudHeader: PointCloudHeader{
+						Fields:    []string{"x", "y", "z"},
+						Size:      []int{4, 4, 4},
+						Count:     []int{1, 1, 1},
+						Type:      []string{"F", "F", "F"},
+						Width:     3,
+						Height:    1,
+						Viewpoint: []float32{1, 2, 3, -0.333, 0.003, 0.895, -0.298},
+					},
+					Points: 3,
+					Data:   make([]byte, 3*4*3),
+				},
+				expectedHeader: PointCloudHeader{
+					Fields:    []string{"x", "y", "z"},
+					Size:      []int{4, 4, 4},
+					Count:     []int{1, 1, 1},
+					Type:      []string{"F", "F", "F"},
+					Width:     3,
+					Height:    1,
+					Viewpoint: []float32{1, 2, 3, -0.333, 0.003, 0.895, -0.298},
+				},
+			},
+		}
+
+		for name, tt := range testCases {
+			tt := tt
+			t.Run(name, func(t *testing.T) {
+				obuf := bytes.NewBufferString("")
```

Comment:
https://pkg.go.dev/bytes#NewBuffer
> In most cases, new(Buffer) (or just declaring a Buffer variable) is sufficient to initialize a Buffer. 
```suggestion
				obuf := &bytes.Buffer{}
```